galeracluster/source/faq.rst

```diff
@@ -5,24 +5,113 @@
 
-.. code-block:: text
+.. rubric:: What is Galera?
+.. _`what-is-galera`:
+
+The word *galera* is the Italian word for *galley*.  The galley is a class of naval vessel used in the Mediterranean Sea from the 2nd millennium :sub:`B.C.E.` until the Renaissance.  Although they used sails when the winds were favorable, their principal method of propulsion came from banks of oars.
+
+In order to manage the vessel effectively, rowers had to act synchronously, lest the oars become intertwined and get blocked.  Captains could scale the crew up to hundreds of rowers, making the galleys faster and more maneuverable in combat.
+
+.. seealso:: For more information on galleys, see `Wikipedia <http://en.wikipedia.org/wiki/Galley>`_.
+
+
+
+.. rubric:: How Do I Manage Failover?
+.. _`failover`:
+
+Galera Cluster is a true synchronous multi-master replication system, which allows the use of any or all of the nodes as master at any time without any extra provisioning.  What this means is that there is no failover in the traditional MySQL master-slave sense.
+
+The primary focus of Galera Cluster is data consistency across the nodes.  This does not allow for any modifications to the database that may compromise consistency.  For instance, the node blocks or rejects write requests until the joining node syncs with the cluster and is ready to process load requests.
+
+The results of this is that you can safely use your favorite approach to distribute or migrate connections between the nodes without the risk of causing inconsistency.
+
+.. seealso:: For more information on connection distribution, see :doc:`deploymentvariants`.
+
+
+
+.. rubric:: How Do I Upgrade the Cluster?
+.. _`faq-upgrade`:
+
+Periodically, updates will become available for Galera Cluster.  The database server itself or the :term:`Galera Replication Plugin`.  To update the software for the node, complete the following steps:
+
+#. Stop the node.
+#. Upgrade the software.
+#. Restart the node.
+
+In addition to this, you also need to transfer client connections from node you want to upgrade to another node for the duration of the migration.
+
+.. seealso:: For more information on upgrade process, see :doc:`upgrading`.
+
+.. rubric:: What InnoDB Isolation Levels does Galera Cluster Support?
+.. _`faq-isolation-levels`:
+
+You can use all isolation levels.  Locally, in a given node, transaction isolation works as it does natively with InnoDB.
+
+That said, globally, with transactions processing in separate node Galera Cluster implements transaction level called ``SNAPSHOT ISOLATION``.  The ``SNAPSHOT ISOLATION`` level occurs between ``REPEATABLE READ`` and ``SERIALIZABLE`` levels.

```

Comment:
"separate nodes", then comma, "implements a transaction level"

"occurs between" -> "is between the"